app/email-preferences/page.tsx

```
@@ -0,0 +1,555 @@
+  const handleConfirmUnsubscribe = () => {
+    // Check if this is a service-level unsubscribe
+    if (confirmDialog.emailId.startsWith('service-')) {
+      const domain = confirmDialog.emailId.replace('service-', '') as EmailDomain;
+
+      // Unsubscribe from all non-required emails in this domain
+      const emailsInDomain = getEmailsByDomain(domain);
+      emailsInDomain.forEach(emailType => {
+        if (!emailType.required) {
+          updatePreference(emailType.id, 'none');
+        }
+      });
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)